sdk/src/main/java/com/sygic/travel/sdk/di/KodeinSetup.kt

```diff
@@ -0,0 +1,108 @@
+package com.sygic.travel.sdk.di
+
+import android.content.Context
+import android.content.pm.PackageInfo
+import android.content.pm.PackageManager
+import android.os.Build
+import com.github.salomonbrys.kodein.Kodein
+import com.github.salomonbrys.kodein.bind
+import com.github.salomonbrys.kodein.instance
+import com.github.salomonbrys.kodein.singleton
+import com.github.salomonbrys.kodein.with
+import com.google.gson.FieldNamingPolicy
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.sygic.travel.sdk.BuildConfig
+import com.sygic.travel.sdk.api.StApi
+import com.sygic.travel.sdk.api.StApiConstants.API_BASE_URL
+import com.sygic.travel.sdk.api.StApiConstants.VERSION_AND_LOCALE
+import com.sygic.travel.sdk.api.interceptors.HeadersInterceptor
+import com.sygic.travel.sdk.api.interceptors.LocaleInterceptor
+import okhttp3.Cache
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+
+
+object KodeinSetup {
+
+	fun setupKodein(context: Context, xApiKey: String) = Kodein {
+		constant("apiKey") with xApiKey
+		bind<Context>() with singleton { context }
+
+		import(networkModule)
+		import(placesModule)
+	}
+
+	private val networkModule = Kodein.Module {
+		constant("cache_size") with 10 * 1024 * 1024L // 10 MB Cache
+
+		bind<Cache>() with singleton {
+			val cacheFile = instance<Context>().cacheDir
+			Cache(cacheFile, instance("cache_size"))
+		}
+
+		bind<Gson>() with singleton {
+			GsonBuilder()
+				.setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+				.create()
+		}
+
+		bind<HttpLoggingInterceptor>("httpIntercept") with singleton {
```

Comment:
Why the name? There could be only one http interceptor. If there will be more, the name is wrong.

Reply:
:heavy_check_mark: